src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingSubmissionResource.java

```diff
@@ -460,12 +458,11 @@ public ResponseEntity<ProgrammingSubmission> getProgrammingSubmissionWithoutAsse
             programmingSubmissionService.checkIfExerciseDueDateIsReached(programmingExercise);
         }
 
-        if (lockSubmission) {
-            Result lockedResult = programmingSubmissionService.lockSubmission(submission, correctionRound);
-            submission = (ProgrammingSubmission) lockedResult.getSubmission();
-        }
-
         if (Objects.nonNull(submission)) {
```

Comment:
nice fix! this was my bad as I remembered an exception from when I was tutor, so I didn't think that that was my fault